assets/_common/js/filters.js

```diff
@@ -0,0 +1,345 @@
+                aVal = a.name?.toLowerCase() || '';
+                bVal = b.name?.toLowerCase() || '';
+                break;
+            case 'projects':
+                aVal = a.projectCount || 0;
+                bVal = b.projectCount || 0;
+                break;
+            case 'modified':
+                aVal = new Date(a.dates?.lastModified || 0);
+                bVal = new Date(b.dates?.lastModified || 0);
+                break;
+            case 'created':
+                aVal = new Date(a.dates?.created || 0);
+                bVal = new Date(b.dates?.created || 0);
+                break;
+            default:
+                return 0;
+        }
+
+        if (direction === 'desc') {
+            return aVal < bVal ? 1 : aVal > bVal ? -1 : 0;
+        }
+        return aVal > bVal ? 1 : aVal < bVal ? -1 : 0;
+    });
+};
+
+const processFilters = (state) => {
+   let filtered = [...state.originalData];
+   
+   filtered = applySearchFilter(filtered, state.filters.search, state.targetType);
+   filtered = applySelectFilters(filtered, state.filters, state.targetType);
+   filtered = sortData(filtered, state.sortBy);
+
+   return {
+    ...state,
+    filteredData: filtered
+   };
+};
+
+const createAgencyCardHTML = (agency) => `
+    <div class="agency-card">
+        <div class="usa-card__container border-base-light radius-0 border-1px hover:shadow-2 card-list-item">
+            <div class="usa-card__media usa-card__media--inset display-block pin-top pin-right">
+                🏥
+            </div>
+            <header class="usa-card__header grid-col-9">
+                <h3 class="usa-card__heading font-heading-lg margin-top-0">
+                    <a href="/agencies/${agency.code}" target="_blank" rel="noopener noreferrer">${ agency.name }</a>
+                </h3>
+            </header>
+            <ul class="width-full usa-card__body font-body-3xs padding-bottom-3 border-bottom-1px border-base-light">
+                <li>
+                    <span class="text-bold">Projects: </span>${agency.projectCount}
+                </li>
+                <li>
+                    <span class="text-bold">Code.JSON: </span>
+                    <a href="#" target="_blank" rel="noopener noreferrer">PLACEHOLDER</a>
+                </li>
+            </ul>
+            <div class="usa-card__footer font-body-3xs padding-bottom-1 padding-top-1px grid-container margin-0">
+                <div class="grid-row margin-bottom-105">
+                    <div class="grid-col-12">
+                        <span class="text-bold"> Organizations: </span>
+                        ${agency.organizations.slice(0, 3).join(', ')}
+                        ${agency.organizations.length > 3 ? `and ${agency.organizations.length - 3} more` : ''}
+                    </div>
+                </div>
+            </div>
+        </div>
+    </div>
+`;
+
+const createProjectCardHTML = (project) => `
+    <div class="project-card">
```

Comment:
If you can fix the errors shown in the html-validation tests, that would be great! https://github.com/DSACMS/code-gov/actions/runs/15407596376/job/43353185736?pr=15